python/tests/integration/helpers.py

```
@@ -332,6 +357,18 @@ def create_server_project(
 ) -> ServerProject:
+    provided = [
+        name
+        for name, value in (
+            ("api_module", api_module),
+            ("api_source", api_source),
+            ("project_api_body", project_api_body),
+        )
+        if value
+    ]
+    if len(provided) > 1:
+        raise ValueError(f"Pass only one of api_module / api_source / project_api_body, got {provided}")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)